pkg/roachpb/data.go

```diff
@@ -1136,6 +1136,11 @@ func CanTransactionRetryAtRefreshedTimestamp(
 			return false, nil
 		}
 	case *WriteTooOldError:
+		// TODO(andrei): Chances of success for on write-too-old conditions might be
+		// usually small: if our txn previously read the key that generated this
```

Comment:
"Might be usually small" seems like a guess. I'd just say that it's possible that we can catch when the refresh is guaranteed to fail and avoid performing it at all in those cases.